test/BuildingRegistry.Tests/Grb/Handlers/GivenGetPreSignedUrlRequest.cs

```diff
@@ -0,0 +1,99 @@
+
+        private readonly PreSignedUrlHandler _handler;
+        private readonly FakeBuildingGrbContext _buildingGrbContext;
+        private readonly Mock<ITicketingUrl> _ticketingUrl;
+        private readonly Mock<ITicketing> _ticketing;
+        private readonly Mock<IAmazonS3Extended> _s3Extended;
+
+        public GivenGetPreSignedUrlRequest()
+        {
+            _fixture = new Fixture();
+
+            _buildingGrbContext = new FakeBuildingGrbContextFactory().CreateDbContext(Array.Empty<string>());
+            _ticketingUrl = new Mock<ITicketingUrl>();
+            _ticketing = new Mock<ITicketing>();
+            _s3Extended = new Mock<IAmazonS3Extended>();
+
+            _handler = new PreSignedUrlHandler(
+                _buildingGrbContext,
+                _ticketing.Object,
+                _ticketingUrl.Object,
+                _s3Extended.Object,
+                Options.Create(new BucketOptions
+                {
+                    BucketName = "Test",
+                    UrlExpirationInMinutes = 60
+                }));
+        }
+
+        [Fact]
+        public async Task ReturnsGetPreSignedUrlResponse()
+        {
+            var ticketId = Guid.NewGuid();
+            var ticketUrl = $"https://api.ticketing.vlaanderen.be/{ticketId}";
+            var preSignedUrl = new Uri("https://signedUrl");
+
+            _ticketing
+                .Setup(x => x.CreateTicket(It.IsAny<IDictionary<string, string>>(), CancellationToken.None))
+                .ReturnsAsync(ticketId);
+            _ticketingUrl
+                .Setup(x => x.For(ticketId))
+                .Returns(new Uri(ticketUrl));
+            _s3Extended
+                .Setup(x => x.CreatePresignedPost(It.IsAny<CreatePresignedPostRequest>()))
+                .Returns(new CreatePresignedPostResponse(preSignedUrl, _fixture.Create<Dictionary<string, string>>()));
+
+            var response = await _handler.Handle(new GetPreSignedUrlRequest(), CancellationToken.None);
+
+            var job = await _buildingGrbContext.Jobs.SingleOrDefaultAsync();
+
+            job.Should().NotBeNull();
+            job!.TicketUrl.Should().Be(ticketUrl);
+            job.Status.Should().Be(JobStatus.Created);
+            response.JobId.Should().Be(job.Id);
+            response.TicketUrl.Should().Be(ticketUrl);
+            response.UploadUrl.Should().Be(preSignedUrl.ToString());
+        }
+
+        [Fact]
+        public async Task WhenError_ThenNoJobIsCreated()
+        {
+            _ticketing
+                .Setup(x => x.CreateTicket(It.IsAny<IDictionary<string, string>>(), CancellationToken.None))
+                .ThrowsAsync(new Exception());
+
+            try
+            {
+                await _handler.Handle(new GetPreSignedUrlRequest(), CancellationToken.None);
+            }
+            catch (Exception)
+            {
+            }
+
+            var job = await _buildingGrbContext.Jobs.SingleOrDefaultAsync();
+
+            job.Should().BeNull();
```

Comment:
This test fails because although the transaction is rolledback, the added job is not discarded. Do we want to explicitly delete the job from the dbcontext or leave as is? I'd expect the db context to be scoped to the request.

Reply:
Yes in the factory you'll need something, maybe a bool, so it creates the same in memory database. So not every time a new guid.